### PR TITLE
feat:[HL-86] Filter By Area API

### PR DESCRIPTION
**Pull request by mhmdatallaa**
#### Proposed Changes

As title

#### Related Issues

N/A

#### Additional Information

N/A

#### Screenshot

N/A

#### Checklist

- [x] I have tested these changes locally.
- [x] I have added appropriate documentation or updated existing documentation.
- [x] I have added appropriate test coverage or updated existing test coverage.
- [ ] I have updated the changelog (if applicable).
- [x] I have followed the project's code style and formatting guidelines.
- [x] I have reviewed and adhered to the project's contributing guidelines.

### CODE REVIEW

**Review by djayyab — APPROVED** [review]

File: Networking/Sources/Networking/Requests/FilterByAreaRequest.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// MARK: - FilterByAreaResponse
+public struct FilterByAreaResponse: Decodable {
+    let meals: [Meal]
+}
+
+// MARK: - Meal
+public struct Meal: Decodable {
+    let id: String
+    let meal: String
+    let mealThumb: String
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Requests/FilterByAreaRequest.swift]:
```suggestion
    public let id: String
    public let meal: String
    public let mealThumb: String
```

File: Networking/Sources/Networking/Requests/FilterByAreaRequest.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import Foundation
+
+// MARK: - FilterByAreaResponse
+public struct FilterByAreaResponse: Decodable {
+    let meals: [Meal]
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Requests/FilterByAreaRequest.swift]:
```suggestion
    public let meals: [Meal]
```

File: Networking/Sources/Networking/Requests/FilterByAreaRequest.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+// MARK: - Meal
+public struct Meal: Decodable {
+    let id: String
+    let meal: String
+    let mealThumb: String
+
+    enum CodingKeys: String, CodingKey {
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Requests/FilterByAreaRequest.swift]:
```suggestion
    private enum CodingKeys: String, CodingKey {
```

File: Healthy/Classes/Entities/FilterByArea.swift [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import Foundation
+
+// MARK: - FilterByAreaResponse
+public struct FilterByArea {
+    let meals: [Meal]
+}
+
```
Comment (ahmdmhasn) [inline Healthy/Classes/Entities/FilterByArea.swift]:
```suggestion
```

**Review by ahmdmhasn — APPROVED** [review]

File: Networking/Sources/Networking/Requests/FilterByAreaRequest.swift [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+// MARK: - FilterByAreaRequest
+public struct FilterByAreaRequest: RequestType {
+    public typealias ResponseType = FilterByAreaResponse
+
+    public init() {}
+
+    public var baseUrl: URL { Constants.theMealDB }
+    public var path: String { "filter.php" }
+    public var method: String = "GET"
+    public var queryParameters: [String: String] {  [ "a": "Canadian" ] }
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Requests/FilterByAreaRequest.swift]: This value should be injected! 
Please check https://github.com/motoon-eg/healthy/pull/93 for an example

**Pull request merged**